driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -229,8 +242,10 @@ public static <T extends Expression> ArrayExpression<T> ofArray(final T... array
     }
 
     /**
-     * Returns an {@linkplain EntryExpression entry} value. An entry is a key
-     * and a value, and is used with {@linkplain MapExpression maps}. An entry
+     * Returns an {@linkplain EntryExpression entry} value. An entry is a
+     * {@linkplain StringExpression string} key and some
+     * {@linkplain Expression value}. Entries are used with
+     * {@linkplain MapExpression maps}. An entry
      * value may be equal to a document value with a field "k" equal to the
      * entry's key and a field "v" equal to the entry's value.
```

Comment:
What does "may be" mean here, is it not always equal?

Reply:
It is vague because pair arrays might, in the future, need to be entries, and we might figure out a way to make it work.

Reply:
I see. Then the question is: how may this information be useful to a user, i.e., what code will a user be able to write given this information that he would not be able to write if not having it, without relying on unspecified API behavior?

Reply:
`some_doc_in_their_database.eq( some_entry )`. If we make the above (hypothetical) changes, this will start to fail, unless we change the contract specified by the docs.

Reply:
[optional]

So I understand that the goal is to tell users that they must not rely on the representation of `EntryExpression` that they may infer via, for example, comparing it with a `Document` via `eq`. If that's correct, wouldn't it be more clear to say so directly instead of telling users the current representation (_"a document value with a field "k" equal to the entry's key and a field "v" equal to the entry's value"_) and implying that they must not rely on it by using the wording _"an entry value **may** be equal to"_? Formally, this wording achieves the goal, which is why I am marking this discussion as optional.

Reply:
I'll remove this sentence.